h2o-extensions/target-encoder/src/main/java/ai/h2o/targetencoding/TargetEncoderHelper.java

```diff
@@ -82,6 +73,90 @@ static double computePriorMean(Frame encodings, int targetClass) {
+   *                          for example when computing interaction for predictions.
+   * @return the index of the interaction column for the group, 
+   * or the index of the column if the group has only one, 
+   * or -1 if one column in the interaction group is missing.
+   */
+  static int createFeatureInteraction(Frame fr, String[] colGroup, String[] interactionDomain) {
+    if (colGroup.length == 1) {
+      return fr.find(colGroup[0]);
+    } else if (new HashSet<>(Arrays.asList(fr.names())).containsAll(Arrays.asList(colGroup))) {
+      return addInteractionColumn(fr, colGroup, interactionDomain);
+    } else { // skip interaction if one col is missing  (we could also replace missing columns with a col of NAs, but we don't do this today when a simple cat column is missing)
+      return -1;
+    }
+  }
+
+  private static int addInteractionColumn(Frame fr, String[] interactingColumns, String[] interactionDomain) {
+    String interactionColName = String.join("~", interactingColumns);  // any limit to col name length?
+    int[] cols = Arrays.stream(interactingColumns).mapToInt(fr::find).toArray();
+    Vec interactionCol = createInteractionColumn(fr, cols, interactionDomain);
+    fr.add(interactionColName, interactionCol);
+    return fr.numCols()-1;
+  }
+  
+  static Vec createInteractionColumn(Frame fr, int[] interactingColumnsIdx, String[] interactionDomain) {
+    String[][] interactingDomains = new String[interactingColumnsIdx.length][];
+    Vec[] interactingVecs = new Vec[interactingColumnsIdx.length];
+    for (int i=0; i<interactingColumnsIdx.length; i++) {
+      Vec vec = fr.vec(interactingColumnsIdx[i]);
+      interactingVecs[i] = vec;
+      interactingDomains[i] = vec.domain();
+    }
+    final InteractionsEncoder encoder = new InteractionsEncoder(interactingDomains, true);
+    byte interactionType = interactionDomain == null ? Vec.T_NUM : Vec.T_CAT;
+    Vec interactionCol = new CreateInteractionTask(encoder, interactionDomain)
+            .doAll(new byte[] {interactionType}, interactingVecs)
+            .outputFrame(null, null, new String[][]{interactionDomain})
+            .lastVec();
+    if (interactionType != Vec.T_CAT)
+      interactionCol = VecUtils.toCategoricalVec(interactionCol); // the domain is obtained from CollectDoubleDomain, so it is sorted by numerical value, and then converted to String
+    return interactionCol;
+  }
+
+  private static class CreateInteractionTask extends MRTask<CreateInteractionTask> {
+    final InteractionsEncoder _encoder;
+    final long[] _interactionDomain; // sorted by construction (see createInteractionColumn), or null
+    
+    public CreateInteractionTask(InteractionsEncoder encoder,  String[] interactionDomain) {
+      _encoder = encoder;
+      _interactionDomain = interactionDomain == null ? null : Arrays.stream(interactionDomain).mapToLong(Long::parseLong).toArray();
```

Comment:
Each node could transfer the num-only domaing from strings to longs locally in the map method and drop it immediately as the `map` function ends.

Cons:
- Each node does the work

Pros: 
- Potentially long domain is not serialized and transfered
- Each node waits for the master node to do the calculation anyway + the serialization time

Reply:
Here, only the `long[]` array is serialized, whereas with your suggestion, the `String[]` array would have to be sent to each node… I believe it's almost always faster with the first array.